code/modules/power/supermatter/supermatter.dm

```diff
@@ -476,10 +478,10 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 				damage = max(damage + (min(removed.temperature - ((T0C + HEAT_PENALTY_THRESHOLD) + (45 * psyCoeff)), 0) / 150 ), 0)
 
 			//Check for holes in the SM inner chamber
-			for(var/t in RANGE_TURFS(1, loc))
-				if(!isspaceturf(t))
+			for(var/turf as anything in RANGE_TURFS(1, loc))
+				if(!isspaceturf(turf))
 					continue
-				var/turf/turf_to_check = t
+				var/turf/turf_to_check = turf
```

Comment:
The same applies here, sorry for missing it. In fact it might be better to do `for(var/turf/open/space/turf_to_check in RANGE_TURFS(1, loc))`, since you do actually want the typechecking behavior